src/test/java/com/google/sps/PastDeeds/PastDeedTesting.java

```diff
@@ -0,0 +1,166 @@
+        pdServlet = new PastDeedsServlet();
+    }
+
+    @Test
+    public void testFetchPosted() {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+
+        Entity postedEntity = new Entity(GOOD_DEED);
+        postedEntity.setProperty(NAME, TITLE);
+        postedEntity.setProperty(DESCRIPTION, DESCRIPTION_INPUT);
+        postedEntity.setProperty(POSTED_YET, TRUE_STRING);
+        postedEntity.setProperty(DAILY_DEED, FALSE_STRING);
+        postedEntity.setProperty(TIMESTAMP, TIMESTAMP_INPUT);
+        postedEntity.setProperty(LINK, LINK);
+        ds.put(postedEntity);
+
+        Entity not_Posted_Entity = new Entity(GOOD_DEED);
+        not_Posted_Entity.setProperty(NAME, TITLE);
+        not_Posted_Entity.setProperty(DESCRIPTION, DESCRIPTION_INPUT);
+        not_Posted_Entity.setProperty(POSTED_YET, FALSE_STRING);
+        not_Posted_Entity.setProperty(DAILY_DEED, FALSE_STRING);
+        not_Posted_Entity.setProperty(TIMESTAMP, TIMESTAMP_INPUT);
+        not_Posted_Entity.setProperty(LINK, LINK);
+        ds.put(not_Posted_Entity);
+
+        GoodDeed posted_deed = 
+            new GoodDeed(postedEntity.getKey(), postedEntity.getKey().getId(), TITLE, DESCRIPTION_INPUT, TRUE, TIMESTAMP_INPUT, LINK);
+        GoodDeed not_posted_deed = 
+            new GoodDeed(not_Posted_Entity.getKey(), not_Posted_Entity.getKey().getId(), TITLE, DESCRIPTION_INPUT, FALSE, TIMESTAMP_INPUT, LINK);
+        
+        List<GoodDeed> actual = pdServlet.fetchPostedDeeds();
+
+        Assert.assertEquals(1, actual.size());
+        Assert.assertEquals(posted_deed.getKey(), actual.get(0).getKey());
+    }
+
+    @Test
+    public void testDoGet() throws IOException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+
+        Entity postedEntity = new Entity(GOOD_DEED);
+        postedEntity.setProperty(NAME, TITLE);
+        postedEntity.setProperty(DESCRIPTION, DESCRIPTION_INPUT);
+        postedEntity.setProperty(POSTED_YET, TRUE_STRING);
+        postedEntity.setProperty(DAILY_DEED, FALSE_STRING);
+        postedEntity.setProperty(TIMESTAMP, TIMESTAMP_INPUT);
+        postedEntity.setProperty(LINK, LINK);
+        ds.put(postedEntity);
+
+        Entity not_Posted_Entity = new Entity(GOOD_DEED);
+        not_Posted_Entity.setProperty(NAME, TITLE);
+        not_Posted_Entity.setProperty(DESCRIPTION, DESCRIPTION_INPUT);
+        not_Posted_Entity.setProperty(POSTED_YET, FALSE_STRING);
+        not_Posted_Entity.setProperty(DAILY_DEED, FALSE_STRING);
+        not_Posted_Entity.setProperty(TIMESTAMP, TIMESTAMP_INPUT);
+        not_Posted_Entity.setProperty(LINK, LINK);
+        ds.put(not_Posted_Entity);
+
+        HttpServletRequest request = mock(HttpServletRequest.class);
+        HttpServletResponse response = mock(HttpServletResponse.class);
+
+        StringWriter sw = new StringWriter();
+        PrintWriter pw = new PrintWriter(sw);
+
+        Mockito.doReturn(pw).when(response).getWriter();
```

Comment:
let's use `thenReturn` instead. Here is the rationale - https://github.com/googleinterns/step118-2020/pull/43#discussion_r459079235

Reply:
also this would usually be `when(response.getWriter()).thenReturn(pw)` -- this way it's more readable.